src/liballoc/tests/slice.rs

```diff
@@ -1643,6 +1648,9 @@ fn panic_safe() {
             }
         }
     }
+
+    // Set default panic hook again.
+    drop(panic::take_hook());
```

Comment:
Setting the global panic hook in a test seems like a bad idea anyway as it affects other tests running in parallel, but this way at least we do not affect tests running later.